src/main/java/mimon/Mimon.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    private static void printExitMessage() {
+        System.out.println(DASHED_LINE);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(DASHED_LINE);
+    }
+
+    private static void listTasks() {
+        System.out.println(DASHED_LINE);
+        System.out.println("Here are the tasks in your list:");
+        if (tasks.isEmpty()) {
+            System.out.println("No tasks added yet.");
+        } else {
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println((i + 1) + ". " + tasks.get(i));
+            }
+        }
+        System.out.println(DASHED_LINE);
+    }
+
+    private static void markTask(String userInput, boolean isDone) throws MimonException {
+        try {
+            int taskIndex = Integer.parseInt(userInput.split(" ")[1]) - 1;
+            if (taskIndex >= 0 && taskIndex < tasks.size()) {
+                if (isDone) {
+                    tasks.get(taskIndex).markAsDone();
+                    System.out.println(DASHED_LINE);
+                    System.out.println("Nice! I've marked this task as done:");
+                } else {
+                    tasks.get(taskIndex).markAsNotDone();
+                    System.out.println(DASHED_LINE);
+                    System.out.println("OK, I've marked this task as not done yet:");
+                }
+                System.out.println("  " + tasks.get(taskIndex));
+                System.out.println(DASHED_LINE);
+            } else {
+                throw new MimonException("Invalid task number. Please enter a valid task number to mark or unmark.");
+            }
+        } catch (NumberFormatException e) {
+            throw new MimonException("Invalid command format. Use: mark/unmark <task_number>");
+        }
+    }
+
+    private static void deleteTask(String userInput) throws MimonException {
+        try {
+            int taskIndex = Integer.parseInt(userInput.split(" ")[1]) - 1;
+            if (taskIndex >= 0 && taskIndex < tasks.size()) {
+                Task removedTask = tasks.remove(taskIndex);
+                System.out.println(DASHED_LINE);
+                System.out.println("Noted. I've removed this task:");
+                System.out.println("  " + removedTask);
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                System.out.println(DASHED_LINE);
+            } else {
+                throw new MimonException("Invalid task number. Please enter a valid task number to delete.");
+            }
+        } catch (NumberFormatException e) {
+            throw new MimonException("Invalid command format. Use: delete <task_number>");
+        }
+    }
+
+    private static void addTodoTask(String description) {
+        addTask(new Todo(description));
+    }
+
+    private static void addDeadlineTask(String input) throws MimonException {
+        String[] parts = input.split(" /by ", 2);
```

Comment:
The name `parts` may not be very intuitive. Consider naming it something like `inputParts` or `splitInput`